text/3-entrees/pallet-coupling.md

```diff
@@ -0,0 +1,141 @@
+```rust, ignore
+/// Checks whether the caller is a member of the set of Account Ids provided by the
+/// MembershipSource type. Emits an event if they are, and errors if not.
+fn check_membership(origin) -> DispatchResult {
+	let caller = ensure_signed(origin)?;
+
+	// Get the members from the vec-set pallet
+	let members = T::MembershipSource::accounts();
+
+	// Check whether the caller is a member
+	ensure!(members.contains(&caller), Error::<T>::NotAMember);
+
+	// If the previous call didn't error, then the caller is a member, so emit the event
+	Self::deposit_event(RawEvent::IsAMember(caller));
+	Ok(())
+}
+```
+
+## Coupling Pallets
+
+Each `check-membership` pallet actually contains very little logic. It has no storage of its own and
+a single extrinsic. Which does the membership checking. All of the heavy lifting is abstracted away
+to another pallet. There are two different ways in which pallets can be coupled to one another, and
+this section investigates both.
+
+### Tight Coupling
+
+The conceptually simpler way of coupling pallets is known as "tight coupling". When tightly coupling
+to another pallet, you specify the depended pallets name explicitly as a trait bound on your
+pallet's configuration trait. This is demonstrated in the tightly-coupled variant of
+`check-membership`.
+
+```rust, ignore
+pub trait Trait: system::Trait + vec_set::Trait {
+	// --snip--
+}
+```
+
+> This pallet, and all pallets, are tightly coupled to frame_system
+
+Supplying this trait bound means that the tightly-coupled variant of `check-membership` pallet can
+only be installed in a runtime that also has the [`vec-set` pallet]() installed. We also see the
+tight coupling in the pallet's `Cargo.toml` file, where `vec-set` is listed by name.
+
+```toml
+vec-set = { path = '../vec-set', default-features = false }
+```
+
+When it comes time to actually get the set of members, we have access to the getter function
+declared in `vec-set`.
+
+```rust, ignore
+// Get the members from the vec-set pallet
+let members = vec_set::Module::<T>::members();
+```
+
+While tightly coupling pallets is conceptually simple, it has the disadvantage that it depends on a
+specific implementation rather than an abstract behavior. The tight version of `check-membership`
+depends on exactly the `vec-set` pallet rather than a behavior such as managing a set of accounts.
+
+## Loose Coupling
+
+Loose coupling solves the problem of coupling to a specific implementation. When loosely coupling to
+another pallet, you add an associated type to the pallets configuration trait, and ensure the
+supplied type implements the necessary behavior by specifying a trait bound.
+
+```rust, ignore
+pub trait Trait: system::Trait {
+	// --snip--
+
+	/// A type that will supply a set of members to check access control against
+	type MembershipSource: AccountSet<AccountId = Self::AccountId>;
+}
+```
+
+Having this associated type means that the losoely-coupled variant of the `check-membership` pallet
```

Comment:
https://english.stackexchange.com/questions/113422/how-to-use-hyphens-appropriately-when-listing-multiple-hyphenated-terms

Reply:
I'm not sure if this relevant because this is not a list of hyphenated items. Even if you decide to leave the hyphen here, please note that the word "loosely" is also misspelled.